web/profiles/custom/os2loop/modules/os2loop_lists/src/Helper/Helper.php

```diff
@@ -0,0 +1,134 @@
+ */
+class Helper {
+  /**
+   * The logged in user account.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The full user entity.
+   *
+   * @var \Drupal\user\UserStorageInterface
+   */
+  protected $userStorage;
+
+  /**
+   * Helper constructor.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entityTypeManager
+   *   Entity type manager.
+   * @param \Drupal\Core\Session\AccountProxyInterface $currentUser
+   *   Logged in user account.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   *   Thrown if the entity type doesn't exist.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   *   Thrown if the storage handler couldn't be loaded.
+   */
+  public function __construct(EntityTypeManagerInterface $entityTypeManager, AccountProxyInterface $currentUser) {
+    $this->userStorage = $entityTypeManager->getStorage('user');
+    $this->currentUser = $currentUser;
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  /**
+   * Get content related by subject.
+   *
+   * @return array
+   *   An array of content that shares subject term with the logged in users
+   *   expertises terms.
+   */
+  public function getContentByUserExpertise(): array {
+    $user_expertises = $this->getCurrentUserExpertisesId();
+    if (empty($user_expertises)) {
+      return [];
+    }
+
+    return $this->entityTypeManager
+      ->getListBuilder('node')
+      ->getStorage()
+      ->loadByProperties([
+        'type' => 'os2loop_question',
+        'status' => 1,
+        'os2loop_shared_subject' => $user_expertises,
+      ]);
+  }
+
+  /**
+   * Get content related by profession.
+   *
+   * @return array
+   *   An array of content that shares profession terms with the logged in
+   *   users professions terms.
+   */
+  public function getContentByUserProfession(): array {
+    $user_professions = $this->getCurrentUserProfessionsId();
+    if (empty($user_professions)) {
+      return [];
+    }
+
+    return $this->entityTypeManager
+      ->getListBuilder('node')
+      ->getStorage()
+      ->loadByProperties([
+        'type' => 'os2loop_question',
+        'status' => 1,
+        'os2loop_shared_profession' => $user_professions,
+      ]);
+  }
+
+  /**
+   * The logged in users expertises.
+   *
+   * @return array
+   *   An array of logged in users expertise taxonomy term ids.
+   */
+  private function getCurrentUserExpertisesId(): array {
+    $term_ids = [];
+    $user = $this->userStorage->load($this->currentUser->id());
+    /** @var \Drupal\user\UserInterface $user */
+    $expertises = $user->get('os2loop_user_areas_of_expertise')->getValue();
+    foreach ($expertises as $expertise) {
+      $term_ids[] = $expertise['target_id'];
+    }
```

Comment:
You could use `array_column($expertises, 'target_id')` here to do the same.

Reply:
Changed

Reply:
Changed